store/storetest/job_store.go

```diff
@@ -360,24 +335,21 @@ func testJobUpdateOptimistically(t *testing.T, ss store.Store) {
 		"Foo": "Bar",
 	}
 
-	if updated, err2 := ss.Job().UpdateOptimistically(job, model.JOB_STATUS_SUCCESS); err2 != nil {
-		if updated {
-			t.Fatal("should have failed due to incorrect old status")
-		}
+	updated, err2 := ss.Job().UpdateOptimistically(job, model.JOB_STATUS_SUCCESS)
+	if updated {
```

Comment:
This is not exactly the same, but I think it should be deterministic, so I think we can remove the `if updated` clause there, and only check for the `require.Nil(t, err)`